libvast/vast/system/explorer.hpp

```diff
@@ -0,0 +1,83 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include "vast/expression.hpp"
+#include "vast/fwd.hpp"
+#include "vast/system/node.hpp"
+#include "vast/type.hpp"
+
+#include <caf/actor.hpp>
+#include <caf/fwd.hpp>
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace vast::system {
+
+using explorer_initial_query_atom = caf::atom_constant<caf::atom("expl_initq")>;
+
+struct explorer_state {
+  static inline constexpr const char* name = "explorer";
+
+  explorer_state(caf::event_based_actor* self);
+
+  // Send the results to the sink, after removing duplicates.
+  void forward_results(vast::table_slice_ptr slice);
+
+  // Size of the time box prior to each event.
+  std::optional<vast::duration> before;
+
+  // Size of the time box after each event.
```

Comment:
```suggestion
  /// Size of the time box after each event.
```

Naming question: Don't we usually just call this a timespan?

Reply:
I think timespan is caf vocabulary, VAST has just `vast::duration`. I've changed it nonetheless, since it sounds like a more standard term than time box.